examples/mises_plasticity/test_plasticity.py

```diff
@@ -0,0 +1,224 @@
+# from matplotlib.ticker import FormatStrFormatter
+from fenics_constitutive import Constraint, IncrSmallStrainProblem
+
+
+def test_uniaxial_strain_3d():
+    mesh = df.mesh.create_unit_cube(MPI.COMM_WORLD, 1, 1, 1)
+    V = df.fem.VectorFunctionSpace(mesh, ("CG", 1))
+    u = df.fem.Function(V)
+    matparam = {
+        "p_ka": 175000,
+        "p_mu": 80769,
+        "p_y0": 1200,
+        "p_y00":2500,
+        "p_w": 200,
+    }
+    law = VonMises3D(matparam)
+
+    def left(x):
+        return np.isclose(x[0], 0.0)
+
+    def right(x):
+        return np.isclose(x[0], 1.0)
+
+    tdim = mesh.topology.dim
+    fdim = tdim - 1
+
+    left_facets = df.mesh.locate_entities_boundary(mesh, fdim, left)
+    right_facets = df.mesh.locate_entities_boundary(mesh, fdim, right)
+    #
+    # ### Dirichlet BCs
+    zero_scalar = df.fem.Constant(mesh, 0.0)
+    scalar_x = df.fem.Constant(mesh, 0.015)
+    fix_ux_left = df.fem.dirichletbc(
+        zero_scalar,
+        df.fem.locate_dofs_topological(V.sub(0), fdim, left_facets),
+        V.sub(0),
+    )
+    move_ux_right = df.fem.dirichletbc(
+        scalar_x,
+        df.fem.locate_dofs_topological(V.sub(0), fdim, right_facets),
+        V.sub(0),
+    )
+
+    dirichlet = [fix_ux_left, move_ux_right]
+    #
+    problem = IncrSmallStrainProblem(law, u, dirichlet)
+
+    solver = NewtonSolver(MPI.COMM_WORLD, problem)
+
+    nTime = 100
+    max_disp = 0.05
+    load_steps = np.linspace(0, 1, num=nTime + 1)[1:]
+    iterations = np.array([], dtype=np.int32)
+    displacement = [0.0]
+    load = [0.0]
+
+    for inc, time in enumerate(load_steps):
+        print("Load Increment:", inc)
+
+        current_disp = time * max_disp
+        scalar_x.value = (current_disp)
+
+        niter, converged = solver.solve(u)
+        problem.update()
+
+        print(f"Converged: {converged} in {niter} iterations.")
+        iterations = np.append(iterations, niter)
+
+        stress_values = []
+        stress_values.append(problem.stress_0.x.array.copy())
+        stress_values = stress_values[0]
+        stress_values = stress_values[::6][0]
+
+        displacement.append(current_disp)
+        load.append(stress_values)
+
+    displacement = np.array(displacement)
+    load = np.array(load)
+
+    # if the maximum stress exceeds the yield limit
+    tolerance = 1e-8
+    assert np.max(load) - matparam["p_y00"] <= tolerance
+
+    # if material behaves linearly under the elastic range with correct slope
+    indices = load + tolerance < matparam["p_y0"]
+    v = (3 * matparam["p_ka"] - 2 * matparam["p_mu"]) / (2 * (3 * matparam["p_ka"] + matparam["p_mu"]))
+    trace = displacement[indices][1] - 2 * v * displacement[indices][1]
+    dev = displacement[indices][1] - trace / 3
+    slope = (matparam["p_ka"] * trace + 2 * matparam["p_mu"] * dev) / displacement[indices][1]
+    assert np.all(abs(np.ediff1d(load[indices]) / np.ediff1d(displacement[indices]) - slope) < 1e-7)
+
+    # ax = plt.subplots()[1]
```

Comment:
These comments should be removed as well

Reply:
Done